src/Microsoft.DotNet.ProjectModel.Loader/LoaderProjectContextExtensions.cs

```diff
@@ -38,7 +38,7 @@ public static AssemblyLoadContext CreateLoadContext(this ProjectContext context,
                 dllImports,
 
                 // Add the project's output directory path to ensure project-to-project references get located
-                new[] { context.GetOutputPathCalculator().GetOutputDirectoryPath(configuration) });
+                new[] { context.GetOutputPathCalculator(configuration).CompilationOutputPath });
```

Comment:
Going back to my comment about naming at the beginning of the PR. This seems weird.